README.md

```diff
@@ -22,26 +23,41 @@ Here is a basic tutorial introduction to monadic parsing with this package.
+A parser turns a string into a data structure. Parsers in this library have the type `Parser s a`, where `s` is the type of the input string, and `a` is the type of the data which the parser will produce on success. `Parser s` is a monad. It’s defined in the module `Text.Parsing.Parser`.
 
-Monads can be used to provide context for a computation, and that’s how we use them in monadic parsing. The context provided by the `Parser` monad is *the parser’s current location in the input string*. Parsing starts at the beginning of the input string.
+Monads can be used to provide context for a computation, and that’s how we use them in monadic parsing.
+The context provided by the `Parser s` monad is __the parser’s current location in the input string__.
+Parsing starts at the beginning of the input string.
 
-Parsing requires two more capabilities: *choice* and *failure*.
+Parsing requires two more capabilities: __alternative__ and __failure__.
 
-We need *choice* to be able to make decisions about what kind of thing we’re parsing depending on the input which we encouter. This is provided by the `Alt` typeclass instance of the `Parser` monad, particularly the `<|>` operator. That operator will first try the left parser and if that fails, then it will backtrack the input string and try the right parser.
+We need __alternative__ to be able to choose what kind of thing we’re parsing depending
+on the input which we encounter. This is provided by the  `<|>` “alt”
+operator of the `Alt` typeclass instance of the `Parser s` monad.
+The expression `p_left <|> p_right` will first try the `p_left` parser and if that fails
+__and consumes no input__ then it will try the `p_right` parser.
 
-We need *failure* in case the input stream is not parseable. This is provided by the `fail` function, which calls the `throwError` function of the `MonadThrow` typeclass instance of the `Parser` monad. The result of running a parser has type `Either ParseError a`, so if the parse succeeds then the result is `Right a` and if the parse fails then the result is `Left ParseError`.
+We need __failure__ in case the input stream is not parseable. This is provided by the `fail`
+function, which calls the `throwError` function of the `MonadThrow` typeclass instance of
+the `Parser s` monad.
 
-
-### Running a parser
-
-To run a parser, call the function `runParser :: s -> Parser s a -> Either ParseError a` in the `Text.Parsing.Parser` module, and supply it with an input string and a parser.
+To run a parser, call the function `runParser :: s -> Parser s a -> Either ParseError a` in
+the `Text.Parsing.Parser` module, and supply it with an input string and a parser.
+If the parse succeeds then the result is `Right a` and if the parse fails then the
```

Comment:
How come there are all these extra line breaks in this file?

Reply:
Just for editing convenience.